src/cpp/rtps/attributes/RTPSParticipantAttributes.cpp

```diff
@@ -224,8 +224,7 @@ static void setup_transports_large_data(
 {
     if (!intraprocess_only)
     {
-        auto shm_transport = create_shm_transport(att, options);
-        att.userTransports.push_back(shm_transport);
+        setup_transports_shm(att, options);
```

Comment:
The log error on `setup_transports_shm` with `FASTDDS_SHM_TRANSPORT_DISABLED` would be incorrect.

The lines below regarding the SHMLocator should also be removed.

So I think it is better to have a setup_large_data_shm_transport with a different error log, and also adding the locator to defaultUnicastLocatorList